From f0ff98ecc33050684efa56f5a51362e88215b557 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 8 Aug 2017 11:04:14 +0800
Subject: [PATCH 2346/5242] MLK-16158-1 video: fbdev: dcss: refine ctxld
 enable code.

commit  ccf1971cdea95086f7d3b8e0b5e942df2d685b88 from
https://source.codeaurora.org/external/imx/linux-imx.git

Refine the ctxld enable configuration by using the
'CTXLD_CTRL_STATUS_SET' to make the code more clean
and strong.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 6330a07..abdd833 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2381,7 +2381,6 @@ static void ctxld_irq_unmask(uint32_t irq_en, struct dcss_info *info)
 static void dcss_ctxld_config(struct work_struct *work)
 {
 	int ret;
-	uint32_t ctxld_ctrl = 0;
 	uint32_t dsb_len, nsgl, esize;
 	struct dcss_info *info;
 	struct platform_device *pdev;
@@ -2425,10 +2424,8 @@ static void dcss_ctxld_config(struct work_struct *work)
 		       info->base + chans->ctxld_addr + CTXLD_DB_COUNT);
 	}
 
-	ctxld_ctrl = readl(info->base + chans->ctxld_addr + CTXLD_CTRL_STATUS);
-	ctxld_ctrl |= (1 << 0);
-
-	writel(ctxld_ctrl, info->base + chans->ctxld_addr + CTXLD_CTRL_STATUS);
+	/* enable ctx_ld */
+	writel(0x1, info->base + chans->ctxld_addr + CTXLD_CTRL_STATUS_SET);
 
 	/* wait finish */
 	reinit_completion(&cfifo->complete);
-- 
1.7.9.5

